### PR TITLE
Key-role-user Logic Introduction 

### PR DESCRIPTION
**Pull request by Sparshkhannaa**
## 📝 Description

This pull request introduces a user role detection feature based on Git contribution analysis. It adds a new module to determine a user's role (e.g., Lead Developer, Contributor) in a project, integrates this information into the analysis pipeline, and displays the detected role in project summaries and bullet points. The implementation is thoroughly tested with a new suite of unit tests.

**User Role Detection and Analysis Integration:**

- Added the `role_detector.py` module, which provides the `detect_user_role` function and `UserRole` dataclass to analyze Git contributions and classify the user's role in a project. It includes logic for weighted contribution calculation and role confidence assessment.
- Integrated user role detection into the project analysis pipeline (`analysis_pipeline.py`). The pipeline now computes the user's role and contribution percentage during analysis and stores this information in the `ProjectAnalysis` object and output dictionaries. 
- Extended the `ProjectAnalysis` dataclass to include `user_role` and `user_contribution_percentage` fields for storing detected role information.

**Role-Aware Output and Display:**

- Updated the TUI rendering logic to display the detected user role and contribution percentage in the project summary if available.
- Enhanced bullet point generation to use role-appropriate action verbs (e.g., "Led development of", "Contributed to") based on the detected role, improving the relevance and accuracy of resume-style project summaries.

**Closes:** #220 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

> Please describe how you tested this PR (both manually and with tests).  

- Added a comprehensive test suite (`test_role_detector.py`) for the user role detection logic, covering various scenarios and edge cases to ensure robust and accurate classification.

- Created a dummy local project and ran this change on that project. 

> Provide instructions so we can reproduce.

uv run --frozen pytest tests/test_role_detector.py tests/test_role_integration.py -v
or
uv run --frozen pytest

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [ ] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [ ] 🔗 Any dependent changes have been merged and published in downstream modules
- [x] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

</details>

### CODE REVIEW

File: src/capstone_project_team_5/role_detector.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        confidence = "High" if user_commits >= 5 else "Medium"
+        return "Major Contributor", confidence
+
+    # Contributor - moderate involvement
+    if contribution_pct >= 10.0:
+        confidence = "Medium" if user_commits >= 3 else "Low"
+        return "Contributor", confidence
+
+    # Minor contributor - small involvement
+    confidence = "Low" if user_commits >= 1 else "Low"
```
Comment (Ron-it) [inline src/capstone_project_team_5/role_detector.py]: both return "Low" ig?

```suggestion
    confidence = "Low"
```

**Review by ojusharma — APPROVED** [review]
Great work! The code is well structured, modularized and testable. It's big feature, so mega brownie points for that!

Just a quick question, should we give the user a choice to change their role in the project? This is a "good-to-have" and can be addressed later if not already addressed.

Overall, incredible work!!
